productcat/src/main/java/com/visenze/productcat/android/ui/PrivacyPolicy.java

```diff
@@ -0,0 +1,138 @@
+package com.visenze.productcat.android.ui;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.CompoundButton;
+import android.widget.TextView;
+
+import com.visenze.productcat.R;
+
+public class PrivacyPolicy {
+
+    private static final String IS_PRIVACY_SHOWN="is_privacy_shown";
+    private static final String IS_TERMS_ACCEPTED="is_terms_accepted";
+    private static final String IS_ADS_ACCEPTED="is_ads_accepted";
+
+    private AlertDialog consentDialog;
+
+    private SharedPreferences pref;
+
+    private boolean isPrivacyShown;
+
+    private boolean isTermsAccepted;
+
+    private boolean isAdsAccepted;
+
+    public PrivacyPolicy(Context context) {
+
+        String pref_file = context.getString(R.string.preference_file);
+        pref = context.getSharedPreferences(pref_file, Context.MODE_PRIVATE);
+
+        isPrivacyShown = pref.getBoolean(IS_PRIVACY_SHOWN, false);
+        isTermsAccepted = pref.getBoolean(IS_TERMS_ACCEPTED, false);
+        isAdsAccepted = pref.getBoolean(IS_ADS_ACCEPTED, false);
+
+        consentDialog = createConsentDialog(context);
+    }
+
+    public void showConsentDialog() {
+        consentDialog.show();
+    }
+
+
+    private AlertDialog createConsentDialog(final Context context) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(context);
+        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+        View view = inflater.inflate(R.layout.dialog_privacy_policy, null);
+
+        final TextView termsAndPolicy = view.findViewById(R.id.terms_and_policy);
+        final CheckBox terms = view.findViewById(R.id.agree_service);
+        terms.setChecked(true);
+
+        final CheckBox ads = view.findViewById(R.id.agree_ad);
+        ads.setChecked(false);
+        final Button acceptBtn = view.findViewById(R.id.btn_accept);
+
+        terms.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean isChecked) {
+                if(isChecked) {
+                    ads.setEnabled(true);
+                    acceptBtn.setEnabled(true);
+                } else {
+                    ads.setEnabled(false);
+                    acceptBtn.setEnabled(false);
+                }
+            }
+        });
+
+
+        builder.setView(view);
+        final AlertDialog dialog = builder.create();
+        dialog.setCanceledOnTouchOutside(false);
+
+        final String policyUrl = context.getString(R.string.weblink_policy);
```

Comment:
we need confirmation on the policy URL , probably should be a page under PDP shopping.visenze.com. Need to talk to See Rong on the URL